Update docs for scheduler ephemeral storage (#4438)
dapr/docs: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md

## Ephemeral Storage

> When running in non-HA mode, Scheduler can be optionally made to use Ephemeral storage, which is in-memory storage that is **not** resilient to restarts, i.e. all Job data will be lost after a Scheduler restart.

Reviewer: ```suggestion
When running in non-HA mode, Scheduler can be optionally made to use Ephemeral storage, which is in-memory storage that is **not** resilient to restarts. For example, all Job data will be lost after a Scheduler restart.
```

> When running in non-HA mode, Scheduler can be optionally made to use Ephemeral storage, which is in-memory storage that is **not** resilient to restarts. For example, all Job data will be lost after a Scheduler restart.

Reviewer 2: ```suggestion
When running in non-HA mode, the Scheduler can be optionally made to use ephemeral storage, which is in-memory storage that is **not** resilient to restarts. For example, all jobs data is lost after a Scheduler restart.
```

> This is useful for non-production deployments where storage is not available or required, or for testing purposes.

Reviewer 2: ```suggestion
This is useful for non-production deployments where storage is not available or required, or for testing purposes.
```
```suggestion
This is useful in non-production deployments or for testing where storage is not available or required.
```
